include/secp256k1.h

```diff
@@ -701,6 +708,188 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_pubkey_combine(
     size_t n
 ) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
 
+/** Opaque data structure that holds a parsed and valid "x-only" public key.
+ *  An x-only pubkey encodes a positive point. That is a point whose Y
+ *  coordinate is a quadratic residue. It is serialized using only its X
+ *  coordinate (32 bytes). A secp256k1_xonly_pubkey is also a secp256k1_pubkey
```

Comment:
> This could be misinterpreted as meaning you can cast a pointer from one to another (which is currently true, but not behavior we should commit to).

Yeah, we do that internally, but shouldn't encourage users to do the same.

> Is it really needed to have support for converting xonly_pubkeys to pubkeys? With a few changes (see below) [...]

I think with the few changes it may still make sense to have the conversion. A user may only have an xonly pubkey but wants to use an existing function. But I have no specific use case in mind. We could add this later if necessary.